src/typed.rs

```diff
@@ -73,7 +73,7 @@ impl<'a> dyn Erased<'a> {
     #[inline]
     pub(crate) fn downcast_mut<'p, T: tags::Type<'p>>(&mut self) -> Option<&mut Tagged<'p, T>> {
         if self.is::<T>() {
-            Some(unsafe { &mut *(self as *mut Self as *mut Tagged<'p, T>) })
+            Some(unsafe { &mut *(self as *mut Self).cast::<Tagged<T>>() })
```

Comment:
Any specific reason for this change? From a pointer provenance perspective this shouldn't make a difference unlike removing the `mem::transmute` you did before, so I'm curious if there are any specific reasons.

Reply:
It was warned by https://rust-lang.github.io/rust-clippy/master/index.html#ptr_as_ptr